Formula/watchexec.rb

```diff
@@ -0,0 +1,18 @@
+class Watchexec < Formula
+  desc "Execute commands when watched files change"
+  homepage "https://github.com/mattgreen/watchexec"
+  url "https://github.com/mattgreen/watchexec/archive/1.6.0.tar.gz"
+  sha256 "0ce7b26bbc05400a835b5a06e13ef69aac90a6d93ef4b989241258494f3c967f"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--release"
+    bin.install "target/release/watchexec"
+    man1.install "doc/watchexec.1"
+  end
+
+  test do
+    system "#{bin}/watchexec", "--version"
```

Comment:
Yeah, testing programs that don't exit is always ungraceful. Your new test looks fine to me under the circumstances, although you should use `assert_match` instead of `assert_equal` (it produces nicer output on failure) and remove the `\n`.